filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala

```diff
@@ -47,17 +62,66 @@ class BlobPathBuilderSpec extends AnyFlatSpec with Matchers{
   }
 
   ignore should "build a blob path from a test string and read a file" in {
-    val endpoint = BlobPathBuilderSpec.buildEndpoint("coaexternalstorage")
-    val endpointHost = BlobPathBuilder.parseURI(endpoint).getHost
+    val storageAccountName = "coaexternalstorage"
+
+    val profile = new AzureProfile(AzureEnvironment.AZURE)
+    val azureCredential = new DefaultAzureCredentialBuilder()
+      .authorityHost(profile.getEnvironment.getActiveDirectoryEndpoint)
+      .build
+    val azure = AzureResourceManager.authenticate(azureCredential, profile).withDefaultSubscription
+
+    // TODO: Can we get the storage account key without all this java-to-scala conversion?
+    val storageAccounts = azure.storageAccounts()
+    val storageAccount = storageAccounts
+      .list()
+      .asScala
+      .find(_.name == storageAccountName)
+      .asInstanceOf[Some[StorageAccount]]
+      .get
```

Comment:
I think maybe this should be `getOrElse` -- seeking advice on how to best handle this whole expression in general

Reply:
So an alternative I could think of to converting to scala could be just using the stream API in java to filter to the correct storage account however would involve most of the same steps and have a java Optional at the end which would still need conversion or unwrapping.

I think keeping with the scala conversion you could try using a match expression here to unwrap the result of 'find' and to perform some of the remaining steps using map (like to produce the first account key)
`    
    val storageAccount = storageAccounts
      .list()
      .asScala
      .find(_.name == storageAccountName)

    val storageAccountKey = storageAccount match {
      case Some(value) => value.getKeys().asScala.map(_.value()).head
      case None => fail("Storage account not found")
    }
`

Reply:
Yeah, unprotected `get` calls are to be avoided, same for `head`. In the final version I think we'll want this whole block to return a `Try` or `ErrorOr` (we can standardize on an error-handling method in the next ticket). 

Here's an example of how the creation of a `Try[BlobServiceClient]` could look, trying to optimize for conciseness rather than readability as a demonstration. (I don't think this is the right final form for this code, just trying to show the kinds of tools that are available.)
```
    val blobServiceClient: Try[BlobServiceClient] = 
      azure
        .storageAccounts()
        .list().asScala
        .find(_.name == storageAccountName).toList
        .flatMap(_.getKeys().asScala)
        .headOption
        .map { k =>
          new BlobServiceClientBuilder()
            .credential(
              new StorageSharedKeyCredential(
                storageAccountName, k.value()
              )
            )
            .endpoint(endpoint)
            .buildClient()
        }
        .map(Success(_))
        .getOrElse(Failure(new Exception("Oh no!")))
```